conformance/lang/expressions/list_constructor.balt

```diff
@@ -0,0 +1,1311 @@
+    io:println(a2 is (true|false)[3]); // @output false
+}
+
+Test-Case: output
+Description: Test inherent type of the created list when the applicable contextually expected type
+             is a list-type-descriptor with different types.
+Labels: list-constructor-expr, is-expr, tuple-type, byte, string, boolean, singleton-type, DecimalNumber
+
+function init() {
+    [int, int, int] t1 = [];
+    io:println(t1 is [int, int, int]); // @output true
+
+    [int...] t2 = [];
+    io:println(t2 is [int...]); // @output true
+
+    [int, int, int] t3 = [1, 2, 3];
+    io:println(t3 is [int, int, int]); // @output true
+    io:println(t3 is [1, 2, 3]); // @output false
+
+    [int...] t4 = [1, 2, 3];
+    io:println(t4 is [int...]); // @output true
+    io:println(t4 is [int, int, int]); // @output false
+    io:println(t4 is [1, 2, 3]); // @output false
+
+    [byte, byte, byte] t5 = [17, 212, 33];
+    io:println(t5 is [byte, byte, byte]); // @output true
+    io:println(t5 is [17, 212, 33]); // @output false
+
+    [byte, byte...] t6 = [17, 212, 33];
+    io:println(t6 is [byte, byte...]); // @output true
+    io:println(t6 is [byte, byte, byte]); // @output false
+    io:println(t6 is [17, 212, 33]); // @output false
+
+    [int, int, string] t7 = [1, 2, "foo"];
+    io:println(t7 is [int, int, string]); // @output true
+    io:println(t7 is [1, 2, "foo"]); // @output false
+
+    [int, [boolean, string], boolean] t8 = [11, [true, "foo"], true];
+    io:println(t8 is [int, [boolean, string], boolean]); // @output true
+    io:println(t8 is [11, [true, "foo"], true]); // @output false
+
+    [int, string...] t9 = [12, "foo", "bar", "baz"];
+    io:println(t9 is [int, string...]); // @output true
+    io:println(t9 is [int, string, string, string]); // @output false
+    io:println(t9 is [12, "foo", "bar", "baz"]); // @output false
+}
+
+Test-Case: output
+Description: Test inherent type of the created list when the applicable contextually expected type
+             is a list-type-descriptor with different types.
+Labels: list-constructor-expr, is-expr, array-type, tuple-type, string, boolean, singleton-type, DecimalNumber
+
+function init() {
+    [int, string][] a1 = [[1, "foo"], [2, "bar"], [3, "baz"]];
+    io:println(a1 is [int, string][]); // @output true
+    io:println(a1 is [int, string][3]); // @output false
+    io:println(a1 is [[1, "foo"], [2, "bar"], [3, "baz"]]); // @output false
+
+    [int, string[2], boolean] t1 = [1, ["foo", "bar"], true];
+    io:println(t1 is [int, string[2], boolean]); // @output true
+    io:println(t1 is [1, ["foo", "bar"], true]); // @output false
+
+    [int, [int, string...], int] t2 = [12, [2, "bar", "qux", "quux"], 21];
+    io:println(t2 is [int, [int, string...], int]); // @output true
+    io:println(t2 is [int, [int, string, string, string], int]); // @output false
+    io:println(t2 is [12, [2, "bar", "qux", "quux"], 21]); // @output false
+}
+
+Test-Case: output
```

Comment:
Shall we add some tests for when just `readonly` is the contextually-expected type also?

See https://github.com/ballerina-platform/ballerina-spec/issues/554.